src/tests/test_cords.cpp

```diff
@@ -0,0 +1,165 @@
+        using namespace config::names;
+        return {{kCsvConfig, csv_config},
+                {kOnlySfd, only_sfd},
+                {kMinCard, minimum_cardinality},
+                {kMaxDiffValsProportion, max_diff_vals_proportion},
+                {kMinSfdStrengthMeasure, min_sfd_strength_measure},
+                {kMinSkewThreshold, min_skew_threshold},
+                {
+                        kMinStructuralZeroesAmount,
+                        min_structural_zeroes_amount,
+                },
+                {kMaxFalsePositiveProbability, max_false_positive_probability},
+                {kCordsDelta, delta},
+                {kMaxAmountOfCategories, max_amount_of_categories}};
+    }
+
+    static std::unique_ptr<algos::Cords> CreateCordsInstance(
+            CSVConfig const& csv_config, bool only_sfd, long double minimum_cardinality = 0.04,
+            long double max_diff_vals_proportion = 0.4, long double min_sfd_strength_measure = 0.3,
+            long double min_skew_threshold = 0.3, long double min_structural_zeroes_amount = 1e-01,
+            long double max_false_positive_probability = 1e-06, long double delta = 0.05,
+            size_t max_amount_of_categories = 70) {
+        return algos::CreateAndLoadAlgorithm<algos::Cords>(GetParamMap(
+                csv_config, only_sfd, minimum_cardinality, max_diff_vals_proportion,
+                min_sfd_strength_measure, min_skew_threshold, min_structural_zeroes_amount,
+                max_false_positive_probability, delta, max_amount_of_categories));
+    }
+};
+
+TEST_F(CordsAlgorithmTest, LineItem) {
+    auto a = CreateCordsInstance(kLineItem, true);
+    a->Execute();
+    AssertVectors(a->GetSoftKeys(), {1, 2, 5, 15});
+    AssertVectors(a->GetTrivialColumns(), {});
+    AssertVectors(a->GetSfd(),
+                  {{0, 8},  {0, 9},  {10, 3},  {11, 3},  {12, 3},  {10, 6},  {11, 6},  {12, 6},
+                   {10, 7}, {11, 7}, {12, 7},  {8, 9},   {10, 8},  {11, 8},  {12, 8},  {10, 9},
+                   {11, 9}, {12, 9}, {10, 13}, {10, 14}, {11, 13}, {11, 14}, {12, 13}, {12, 14}});
+}
+
+TEST_F(CordsAlgorithmTest, iris) {
+    auto a = CreateCordsInstance(kiris, true);
+    a->Execute();
+    AssertVectors(a->GetSoftKeys(), {});
+    AssertVectors(a->GetTrivialColumns(), {});
+    AssertVectors(a->GetSfd(), {{2, 4}, {3, 4}});
+}
+
+TEST_F(CordsAlgorithmTest, CIPublicHighway10k) {
+    auto a = CreateCordsInstance(kCIPublicHighway10k, true);
+    a->Execute();
+    AssertVectors(a->GetSoftKeys(), {0, 2});
+    AssertVectors(a->GetTrivialColumns(),
+                  {1,  7,  8,  9,  10, 14, 15, 16, 17, 7,  8,  9,  10, 14, 15, 16, 17, 7,
+                   8,  9,  10, 14, 15, 16, 17, 7,  8,  9,  10, 14, 15, 16, 17, 7,  8,  9,
+                   10, 14, 15, 16, 17, 14, 15, 16, 17, 14, 15, 16, 17, 14, 15, 16});
+    AssertVectors(a->GetSfd(), {{11, 3}, {11, 4}, {11, 5}, {11, 6}, {11, 12}});
+}
```

Comment:
I wrote, but forgot to attach a comment with motivation
1) I don't like using default parameters. When reading the tests themselves, which call functions that use these default parameters, it is awkward to go back to the function definition to find out the default values.
2) I don’t like such functions, since calling them turns into passing a set of numbers, the meaning of which is difficult to understand (how to map values to param names).
3) We have to duplicate parameters with their types in `GetParamMap` and `CreateCordsInstance`
4) At the moment, we need the following from these functions: we want to be able to set a config for several tests and use it. But also later we want to be able to add other tests that can use other configs.

Moreover, these problems may not be obvious when we have several parameters, but in this case there are too many of them.
Therefore, I would like to propose the following solution: let's separate the config into a structure (there is no table in the config, since we want to be able to reuse the same test config for different tables). And we also use Designated initializers when creating the config, i.e. We explicitly indicate the names of the fields.
Thanks to this, we solve the problems that I wrote about above.